Content.Server/Anomaly/Effects/ChasingAnomalySystem.cs

```diff
@@ -0,0 +1,79 @@
+using System.Linq;
+using System.Numerics;
+using Content.Server.Anomaly.Components;
+using Content.Shared.Anomaly.Components;
+using Robust.Shared.Physics.Systems;
+using Robust.Shared.Random;
+
+namespace Content.Server.Anomaly.Effects;
+/// <summary>
+/// This component allows the anomaly to chase a random instance of the selected type component within a radius.
+/// </summary>
+public sealed class ChasingAnomalySystem : EntitySystem
+{
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly SharedPhysicsSystem _physics = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<ChasingAnomalyComponent, AnomalyPulseEvent>(OnPulse);
+    }
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        foreach (var (anom, trans) in EntityManager.EntityQuery<ChasingAnomalyComponent, TransformComponent>(true))
+        {
+            if (!Initialized(anom.Owner)) return;
+            if (Deleted(anom.ChasingEntity)) continue;
+            if (!anom.ChasingEntity.IsValid()) continue;
+            if (anom.ChasingEntity == default!) continue;
+
+            //Calculating direction to the target.
+            var xformQuery = GetEntityQuery<TransformComponent>();
+            var xform = xformQuery.GetComponent(anom.ChasingEntity);
+
+            var currPos = new Vector2(trans.MapPosition.X, trans.MapPosition.Y);
+            var targetPos = new Vector2(xform.MapPosition.X, xform.MapPosition.Y);
+            var delta = targetPos - currPos;
+
+            //Avoiding "shaking" when the anomaly approaches close to the target.
+            if (delta.Length() < 1f) continue;
+
+            //Changing the direction of the anomaly.
+            var speed = delta.Normalized() * anom.CurrentSpeed;
+            _physics.SetLinearVelocity(anom.Owner, speed);
+        }
+
+    }
+
+    private void OnPulse(EntityUid uid, ChasingAnomalyComponent component, ref AnomalyPulseEvent args)
+    {
+        //Speed updating.
+        component.CurrentSpeed = args.Severity * component.MaxChasingSpeed;
+
+        //Speed up on supercritical
+        if (args.Severity >= 1)
+            component.CurrentSpeed *= component.SuperCriticalSpeedModifier;
+
+        //We find our coordinates and calculate the radius of the target search.
+        var xformQuery = GetEntityQuery<TransformComponent>();
+        var xform = xformQuery.GetComponent(uid);
+        var range = component.MaxChaseRadius * args.Severity;
+        var compType = EntityManager.ComponentFactory.GetRegistration(component.ChasingComponent).Type;
```

Comment:
idk if this is bad someone that knows about component registries and lookups should check it

Reply:
@deltanedas 
i use ProximityBeeper part of code
https://github.com/space-wizards/space-station-14/blob/bcb79eff4b78f1a7d8fa726a9b9b8b612971e0c4/Content.Server/Pinpointer/ProximityBeeperSystem.cs#L65

Reply:
pinpointer is probably still old so idk anyway a maint should look